### PR TITLE
Docs/VSCodeConfiguration: Resolve all FIXMEs related to clang compiler

### PR DESCRIPTION
**Pull request by tbhaxor**

### CODE REVIEW

File: Meta/serenity.sh [hunk 1]
```diff
@@ -101,7 +101,8 @@ if [ "$TARGET" != "lagom" ] && [ -n "$1" ]; then
 else
     TOOLCHAIN_TYPE="${SERENITY_TOOLCHAIN:-"GNU"}"
 fi
-if ! [[ "${TOOLCHAIN_TYPE}" =~ ^(GNU|Clang)$ ]]; then
+[[ "${TOOLCHAIN_TYPE}" = "clang" ]] && TOOLCHAIN_TYPE='Clang'
+if ! [[ "${TOOLCHAIN_TYPE}" =~ ^(GNU|Clang|clang)$ ]]; then
```
Comment (BertalanD) [inline Meta/serenity.sh:105]: Why is this necessary?

Comment (tbhaxor) [reply Meta/serenity.sh]: This comes from the tasks.json. The build directory is `Build/x86_64clang`. From the input value I am appending the compiler name to the build directory and also passing that same value to the `Meta/serenity.sh run` command-line. 

There are only two valid compilers `Clang` and `GNU`. So if the compiler is _clang_, I am transforming it to Clang so that it is considered valid.

Comment (tbhaxor) [reply Meta/serenity.sh]: If you remove it, then it will fail as invalid compiler. And making it case-insensitive everywhere doesn't make sense and could be error-prone.

Comment (tbhaxor) [reply Meta/serenity.sh]: Hi @BertalanD, Could you please re-review the PR? Also I have removed `clang` from line 105 in b94c1733550ee34b50ee96def3f358f579f93412
